lib/Parser/Parse.cpp

```diff
@@ -4505,48 +4486,21 @@ ParseNodePtr Parser::ParseFncDecl(ushort flags, LPCOLESTR pNameHint, const bool
     *m_ppnodeVar = nullptr;
     m_ppnodeVar = ppnodeVarSave;
 
-    // Restore the current function.
-    if (buildAST)
-    {
-        Assert(pnodeFnc == m_currentNodeFunc);
-
-        m_currentNodeFunc = pnodeFncSave;
-        m_pCurrentAstSize = pAstSizeSave;
-
-        if (!fLambda)
-        {
-            Assert(pnodeFnc == m_currentNodeNonLambdaFunc);
-            m_currentNodeNonLambdaFunc = pnodeFncSaveNonLambda;
-        }
-    }
-    else
-    {
-        Assert(pnodeFnc == m_currentNodeDeferredFunc);
-        if (!fLambda)
-        {
-            Assert(pnodeFnc == m_currentNodeNonLambdaDeferredFunc);
-            m_currentNodeNonLambdaDeferredFunc = pnodeFncSaveNonLambda;
-        }
-        m_currentNodeDeferredFunc = pnodeFncSave;
-        if (m_currentNodeFunc && pnodeFnc->sxFnc.HasWithStmt())
-        {
-            GetCurrentFunctionNode()->sxFnc.SetHasWithStmt(true);
-        }
-    }
+    // Lambdas do not have "arguments" and instead capture their parent's
+    // binding of "arguments.  To ensure the arguments object of the enclosing
+    // non-lambda function is loaded propagate the UsesArguments flag up to
+    // the parent function
```

Comment:
Why did this comment move up here?  It doesn't mention eval at all, and this code here isn't specific to lambdas